src/main/groovy/wooga/gradle/unity/version/manager/UnityVersionManagerPlugin.groovy

```diff
@@ -124,70 +126,127 @@ class UnityVersionManagerPlugin implements Plugin<Project> {
                         true
                     })
                     .collect({
-                BuildTargetToComponent.buildTargetToComponent(it.buildTarget)
-            })
-        }).flatMap(new Transformer<Provider<List<Component>>, List<Provider<Component>>>() {
+                        BuildTargetToComponents.buildTargetToComponents(it.buildTarget, extension.unityVersion)
+                    })
+        }).flatMap(new Transformer<Provider<List<Component>>, List<Provider<List<Component>>>>() {
             @Override
-            Provider<List<Component>> transform(List<Provider<Component>> providers) {
+            Provider<List<Component>> transform(List<Provider<List<Component>>> providers) {
                 project.provider({
-                    providers.collect {
-                        it.getOrElse(Component.unknown)
-                    }.findAll {it != Component.unknown }
+                    providers.collectMany {it.getOrElse([]) }
                 })
             }
         }))
         extension
     }
 
+    @Deprecated
     static Component buildTargetToComponent(BuildTarget target) {
         buildTargetToComponent(target.toString())
     }
 
+    @Deprecated
     static Component buildTargetToComponent(String target) {
-        Component component = Component.unknown
+        buildTargetToComponents(target, "2017.1.1f1").first()
+    }
+
+    static List<Component> buildTargetToComponents(String target, String versionString) {
+        def version = new DefaultArtifactVersion(versionString.split(/f|p|b|a/).first().toString())
+        def osName = System.getProperty("os.name").toLowerCase()
+        def isWindows = osName.contains("windows")
+        def isLinux = osName.contains("linux")
+        def isMac = osName.contains("mac os")
```

Comment:
We have helper classes in gradle-commons.

https://github.com/wooga/gradle-commons/blob/master/src/main/groovy/com/wooga/gradle/PlatformUtils.groovy

Reply:
ahh nice. Will use them. Thanks